fetchai/ledger/serialisation/transaction.py

```diff
@@ -141,7 +142,25 @@ def encode_payload(buffer: io.BytesIO, payload: transaction.Transaction):
         identity.encode(buffer, signer)
 
 
-def encode_transaction(payload: transaction.Transaction, signers: List[crypto.Entity]):
+def encode_multisig_transaction(payload: 'Transaction', signatures: Dict[Identity, bytes]):
+    assert isinstance(payload, bytes) or isinstance(payload, transaction.Transaction)
+
+    # encode the contents of the transaction
+    buffer = io.BytesIO()
+    encode_payload(buffer, payload)
+
+    # append signatures in order
+    for signer in payload.signers.keys():
+        if isinstance(signatures[signer], bytes):
+            bytearray.encode(buffer, signatures[signer])
+        else:
+            bytearray.encode(buffer, signatures[signer]['signature'])
+
+    # return the encoded transaction
+    return buffer.getvalue()
+
+
+def encode_transaction(payload: 'Transaction', signers: List[crypto.Entity]):
```

Comment:
I agree with this - currently there is a separation between the serialization code and the crypto objects, I think it would actually be a lot cleaner if the Transaction object could serialize itself. I'll do a little refactoring of this if possible now.